### ISSUE TITLE
PDP evaluation status

### ISSUE THREAD

**Issue by tr33** [openid/authzen#52]
A PDP may involve third party components into policy evaluation as a sub-query, i.e. a Policy Information Point (PIP) which holds additional information about the resource/subjects relationships or other metadata required for the policy.

If one of those sub-queries fail (e.g. because the PIP is down or the query could not be processed correctly), the PDP should communicate this situation to the PEP.

there should be a response scheme by the PDP for the PEP, where the PDP differenciates between "_yes, I have understood your decision request and here is your response for enforcement_" and "_I have trouble to decide on your request, so my response would be neither a 'yes' or 'no'_"

Proposal wording for discussion:

```
If successful, the PDP returns a `200 OK` response code and an evaluation result document in the response body.
A successful response **always** contains the following elements:

* status: indicates whether the request was processed without errors and contains a valid authorization decision, OR if there were any errors. Possible values:
    * "success": authorization request successfully evaluated, no errors.
    * "error": Some errors occurred during the evaluation or the input parameters were faulty. In this case, an element "error" is included.
* allow: <true/false> gives the PDPs policy decision. ```true``` indicates "permission granted", ```false``` indicates "permission denied".
```

**Comment by tr33** [comment]
this issue can be postponed to later discussion

**Comment by ggebel** [comment]
This issue will remain open until the group decides on overall structure of API for request and response formats.

**Comment by identitymonk** [comment]
I disagree with @tr33 proposal here. The response already has a `decision` attribute to indicate that the response was well received, understood, and accepted. While the code might be of family `2XX`, if there is a problem in the backend then it cannot be a `200`. 

HTTP defines plenty of other codes to handle this kind of situation, which from the PEP endpoint might be an easier way to discriminate what is happening.

**Comment by ogazitt** [comment]
I believe we already specify enough behavior [here](https://openid.github.io/authzen/#name-error-responses).

If the PDP was able to evaluate the request, it returns a `200` with a `decision` (true or false).

If the PDP could not process the request, we have a set of error conditions defined in the HTTP binding.
